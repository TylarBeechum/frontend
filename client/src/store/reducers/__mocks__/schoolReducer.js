import { createContext } from 'react';

export const SCHOOL_DATA_START = 'SCHOOL_DATA_START';
export const SCHOOL_DATA_SUCCESS = 'SCHOOL_DATA_SUCCESS';
export const SCHOOL_DATA_ERROR = 'SCHOOL_DATA_ERROR';
export const SEARCH_HANDLE_CHANGE = 'SEARCH_HANDLE_CHANGE';
export const REMOVE_CREDENTIAL_START = 'REMOVE_CREDENTIAL_START';
export const REMOVE_CREDENTIAL_SUCCESS = 'REMOVE_CREDENTIAL_SUCCESS';
export const REMOVE_CREDENTIAL_ERROR = 'REMOVE_CREDENTIAL_ERROR';
export const UPDATE_CRED_DATA = 'UPDATE_CRED_DATA';

export const schoolContext = createContext();

const initialState = {
  schoolData: {
    schoolDetails: {
      name: 'testName',
      credentials: [
        {
          id: 1,
          credName: 'test1',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        },
        {
          id: 2,
          credName: 'test2',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        },
        {
          id: 3,
          credName: 'test3',
          criteria: 'test',
          ownerName: 'Nathan Thomas'
        }
      ]
    }
  },
  schoolDataSuccess: true,
  schoolDataStart: false,
  schoolDataError: false,
  schoolSearchInput: '',
  removeCredentialStart: false,
  removeCredentialSuccess: false,
  removeCredentialError: false
};

export const schoolReducer = (state = initialState, action) => {
  switch (action.type) {
    case SCHOOL_DATA_START:
      return {
        ...state,
        schoolDataStart: true,
        schoolDataSuccess: false,
        schoolDataError: false
      };
    case SCHOOL_DATA_SUCCESS:
      return {
        ...state,
        schoolDataStart: false,
        schoolDataSuccess: true,
        schoolDataError: false,
        schoolData: action.payload.data.getUserById
      };
    case SCHOOL_DATA_ERROR:
      return {
        ...state,
        schoolDataError: true,
        schoolDataStart: false
      };
    case SEARCH_HANDLE_CHANGE:
      return {
        ...state,
        schoolSearchInput: action.payload
      };
    case REMOVE_CREDENTIAL_START:
      return {
        ...state,
        removeCredentialStart: true,
        removeCredentialSuccess: false,
        removeCredentialError: false
      };
    case REMOVE_CREDENTIAL_SUCCESS:
      return {
        ...state,
        removeCredentialStart: false,
        removeCredentialSuccess: true,
        removeCredentialError: false,
        schoolData: {
          ...state.schoolData,
          schoolDetails: {
            ...state.schoolData.schoolDetails,
            credentials: state.schoolData.schoolDetails.credentials.filter(
              credential => credential.id !== action.payload.credId
            )
          }
        }
      };
    case REMOVE_CREDENTIAL_ERROR:
      return {
        ...state,
        removeCredentialStart: false,
        removeCredentialSuccess: false,
        removeCredentialError: true
      };
    case UPDATE_CRED_DATA:
      return {
        ...state,
        schoolData: {
          ...state.schoolData,
          schoolDetails: {
            ...state.schoolData.schoolDetails,
            credentials: action.payload
          }
        }
      };
    default:
      return state;
  }
};
