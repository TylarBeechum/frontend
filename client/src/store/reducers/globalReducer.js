import { createContext } from 'react';

export const REGISTER = 'REGISTER';
export const HANDLE_CRED_CHANGES = 'HANDLE_CRED_CHANGES';
export const ON_BOARD_DETAILS = 'ON_BOARD_DETAILS';
export const RESET_CREDENTIAL_FORM = 'RESET_CREDENTIAL_FORM';
export const globalContext = createContext();

const initialState = {
  user: null,
  ownerName: '',
  credName: '',
  description: '',
  type: '',
  studentEmail: '',
  imageUrl: '',
  criteria: '',
  issuedOn: '',
  expirationDate: '',
  schoolId: localStorage.id
};

export const globalReducer = (state = initialState, action) => {
  switch (action.type) {
    case REGISTER:
      return {
        ...state,
        user: action.payload
      };
    case ON_BOARD_DETAILS:
      return state;
    case HANDLE_CRED_CHANGES:
      return {
        ...state,
        [action.payload.name]: action.payload.value
      };
    case RESET_CREDENTIAL_FORM:
      return {
        ...state,
        ownerName: '',
        credName: '',
        description: '',
        type: '',
        studentEmail: '',
        imageUrl: '',
        criteria: '',
        issuedOn: '',
        expirationDate: '',
        schoolId: localStorage.id
      };
    default:
      return state;
  }
};

export default globalReducer;
