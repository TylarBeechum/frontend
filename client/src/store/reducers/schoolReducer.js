import { createContext } from 'react';

export const schoolContext = createContext();

const initialState = {};

export const schoolReducer = (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};
