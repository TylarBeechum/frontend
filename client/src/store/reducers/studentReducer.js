import { createContext } from 'react';

export const STUDENT_DATA_START = 'STUDENT_DATA_START';
export const STUDENT_DATA_SUCCESS = 'STUDENT_DATA_SUCCESS';
export const STUDENT_DATA_ERROR = 'STUDENT_DATA_ERROR';
export const SEARCH_HANDLE_CHANGE = 'SEARCH_HANDLE_CHANGE';
export const SET_STUDENT_DATA = 'SET_STUDENT_DATA';
export const STUDENT_EMAIL_UPDATE = 'STUDENT_EMAIL_UPDATE';
export const REMOVE_STUDENT_EMAIL = 'REMOVE_STUDENT_EMAIL';

export const studentContext = createContext();

const initialState = {
  studentData: null,
  studentDataSuccess: false,
  studentDataStart: false,
  studentDataError: false,
  studentSearchInput: ''
};

export const studentReducer = (state = initialState, action) => {
  console.log(action);
  switch (action.type) {
    case STUDENT_DATA_START:
      return {
        ...state,
        studentDataStart: true,
        studentDataSuccess: false,
        studentDataError: false
      };
    case STUDENT_DATA_SUCCESS:
      return {
        ...state,
        studentDataStart: false,
        studentDataSuccess: true,
        studentDataError: false,
        studentData: action.payload.data.getUserById
      };
    case STUDENT_DATA_ERROR:
      return {
        ...state,
        studentDataError: true,
        studentDataStart: false
      };
    case SEARCH_HANDLE_CHANGE:
      return {
        ...state,
        studentSearchInput: action.payload
      };
    case SET_STUDENT_DATA:
      return {
        ...state,
        studentData: {
          ...state.studentData,
          studentDetails: action.payload.data.addStudentDetail
        },
        studentDataSuccess: true
      };
    case STUDENT_EMAIL_UPDATE:
      const { email, id, userId } = action.payload;
      return {
        ...state,
        studentData: {
          ...state.studentData,
          studentDetails: {
            ...state.studentData.studentDetails,
            emailList: [
              ...state.studentData.studentDetails.emailList,
              { email, id, userId }
            ],
            credentials: [
              ...state.studentData.studentDetails.credentials,
              ...action.payload.credentials
            ]
          }
        }
      };
    case REMOVE_STUDENT_EMAIL:
      return {
        ...state,
        studentData: {
          ...state.studentData,
          studentDetails: {
            ...state.studentData.studentDetails,
            emailList: action.payload.emailList,
            credentials: action.payload.credList
          }
        }
      };
    default:
      return state;
  }
};
