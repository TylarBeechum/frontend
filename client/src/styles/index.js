import { createGlobalStyle } from 'styled-components';

import Reset from './Reset';
import Global from './Global';

const GlobalStyle = createGlobalStyle`
  ${Reset}
  
  ${Global}

  background-color: red;
`;

export default GlobalStyle;
