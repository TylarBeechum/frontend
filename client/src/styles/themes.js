import styled from 'styled-components';
import { Button, Form, TextInput, FormField } from 'grommet';

export const theme = {
  global: {
    colors: {
      // primary: '#444444',
      // secondary: 'palevioletred',
      fontColor: '#333333',
      bodyBackground: '#f8f8f8',
      teal: 'accent-1',
      dashBoardBorder: '#adadad',
      dashBoardBg: '#ffffff',
      btnHoverColor: '#81FCED',
      focus: 'accent-4',
      navbarHoverBg: '#ffffff',
      searchBarBorder: '#adadad',
      searchBarColor: '#adadad',
      dashboardNotFoundColor: '#adadad',
      mobileTrashBtnBorder: '#adadad',
      confirmationInputBorder: '#adadad',
      footerColor: '#adadad'
    },
    border: '1px solid rgba(0,0,0,0.33)',
    // fontFamily
    radius: '6px',
    font: {
      family: 'Roboto'
    }
  },
  tab: {
    active: {
      color: 'brand'
    },
    color: '#333333',
    border: {
      color: '#f8f8f8',
      active: {
        color: 'brand'
      }
    }
  }
};

export const BaseButton = styled(Button)``;

export const SecondaryButton = styled(Button)`
  color: white;
  border: 1px solid white;
  :hover {
    border: 2px solid white;
  }
`;

export const BaseForm = styled(Form)``;

export const BaseTextInput = styled(TextInput)``;

export const BaseFormField = styled(FormField)``;
