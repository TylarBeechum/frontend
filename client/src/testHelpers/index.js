import renderWithRouter from './renderWithRouter/renderWithRouter';
import rendererWithRouter from './rendererWithRouter/rendererWithRouter';

export { renderWithRouter, rendererWithRouter };
