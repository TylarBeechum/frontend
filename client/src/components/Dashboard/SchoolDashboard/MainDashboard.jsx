import React, { useEffect } from 'react';
import { useStateValue } from 'react-conflux';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { InfiniteScroll, Box } from 'grommet';
import FuzzySearch from 'fuzzy-search';

import { BaseButton } from '../../../styles/themes';
import searchIcon from '../../../images/search-icon.svg';

import CredCard from '../Card/CredCard';
import DashboardLoading from '../DashboardLoading';

import queries from './queries';
import {
  schoolContext,
  SCHOOL_DATA_START,
  SCHOOL_DATA_SUCCESS,
  SCHOOL_DATA_ERROR,
  SEARCH_HANDLE_CHANGE
} from '../../../store/reducers/schoolReducer';

const MainDashboard = ({ history }) => {
  const [state, dispatch] = useStateValue(schoolContext);
  useEffect(() => {
    dispatch({ type: SCHOOL_DATA_START });
    async function getUserData() {
      try {
        const { id } = localStorage;
        const data = await queries.getUserById({
          id
        });
        dispatch({ type: SCHOOL_DATA_SUCCESS, payload: data });
      } catch (err) {
        dispatch({ type: SCHOOL_DATA_ERROR });
      }
    }
    getUserData();
  }, [dispatch]); // Re-render whenever an action in schoolContext is dispatched
  let searchResult = [];
  if (state.schoolData) {
    const searchTerms = ['credName', 'criteria', 'ownerName', 'issuedOn'];
    const searchOptions = {
      caseSensitive: false
    };
    const searcher = new FuzzySearch(
      state.schoolData.schoolDetails.credentials,
      searchTerms,
      searchOptions
    );
    searchResult = searcher.search(state.schoolSearchInput);
  }
  const handleChange = e => {
    dispatch({ type: SEARCH_HANDLE_CHANGE, payload: e.target.value });
  };
  return (
    <>
      <SchoolDetails>
        {state.schoolDataSuccess ? (
          <h2>{state.schoolData.schoolDetails.name}</h2>
        ) : (
          <div />
        )}
        <div>
          <input
            type="text"
            name="searchText"
            placeholder="Search"
            onChange={handleChange}
            value={state.schoolSearchInput}
          />
          <IssueCredButton
            type="button"
            onClick={() => history.push('/dashboard/credForm')}
            label="+ Issue Credential"
            primary
          />
        </div>
      </SchoolDetails>
      {state.schoolDataStart ? (
        <DashboardLoading />
      ) : (
        <>
          {state.schoolDataSuccess && searchResult.length ? (
            <Box height="75vh" overflow="auto">
              <InfiniteScroll items={searchResult} step={10}>
                {item => {
                  return <CredCard key={item.id} cred={item} />;
                }}
              </InfiniteScroll>
            </Box>
          ) : (
            state.schoolDataSuccess && (
              <NothingFound>No results were found...</NothingFound>
            )
          )}
        </>
      )}
    </>
  );
};

MainDashboard.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const SchoolDetails = styled.section`
  margin: 50px auto 30px;
  max-width: 1600px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;

  div {
    display: flex;
    width: 68%;
    justify-content: flex-end;
    align-items: center;

    input {
      background: white url(${searchIcon}) no-repeat scroll 5px 5px;
      background-size: 20px;
      background-position: right 13px center;
      border: 2px solid ${({ theme }) => theme.global.colors.searchBarBorder};
      color: ${({ theme }) => theme.global.colors.searchBarColor};
      height: 40px;
      margin-right: 1.5%;
      padding: 25px 5% 25px 2.5%;
      transition: 0.25s ease-in-out;
      width: 220px;
      caret-color: grey;
      border-radius: 50px;
      font-size: 1.8rem;

      ::placeholder {
        color: ${({ theme }) => theme.global.colors.searchBarColor};
        font-weight: 700;
      }

      &:focus {
        color: ${({ theme }) => theme.global.colors.brand};
        border-color: ${({ theme }) => theme.global.colors.brand};
        outline: none;
        padding-left: 15px;
        width: 50%;
      }
    }
  }
`;

const IssueCredButton = styled(BaseButton)`
  padding: 12px 15px;
  color: white;
  text-align: right;
  border-radius: 50px;
  margin-left: 2%;
`;

const NothingFound = styled.p`
  width: 100%;
  text-align: center;
  font-size: 2.4rem;
  margin-top: 20vh;
  color: ${({ theme }) => theme.global.colors['status-disabled']};
`;

const CredListContainer = styled(Box)`
  section:last-of-type {
    margin-bottom: 15px;
  }
`;

export default MainDashboard;
