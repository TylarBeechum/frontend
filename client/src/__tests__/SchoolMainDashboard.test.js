/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { StateProvider } from 'react-conflux';
import { Grommet } from 'grommet';
import { theme } from '../styles/themes';
import { renderWithRouter, rendererWithRouter } from '../testHelpers';

import MainDashboard from '../components/Dashboard/SchoolDashboard/MainDashboard';
import { schoolReducer, schoolContext } from '../store/reducers/schoolReducer';

describe('<MainDashboard /> within school', () => {
  describe('tests that <MainDashboard /> within school renders without crashing', () => {
    it('should renders the <MainDashboard /> component without crashing', () => {
      const helpers = renderWithRouter(
        <StateProvider reducer={schoolReducer} stateContext={schoolContext}>
          <Grommet theme={theme}>
            <MainDashboard history={history} />
          </Grommet>
        </StateProvider>
      );
    });
  });
  describe('tests <MainDashboard /> within school matches previous snapshot', () => {
    it('should matche the snapshot of Main Dashboard', () => {
      const tree = rendererWithRouter(
      <StateProvider reducer={schoolReducer} stateContext={schoolContext}>
          <Grommet theme={theme}>
            <MainDashboard history={history} />
          </Grommet>
        </StateProvider>
      )
        expect(tree.toJSON()).toMatchSnapshot();
    })
  })
});
