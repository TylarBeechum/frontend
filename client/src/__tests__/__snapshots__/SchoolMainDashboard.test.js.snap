// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainDashboard /> within school tests <MainDashboard /> within school matches previous snapshot should matche the snapshot of Main Dashboard 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <section
    className="sc-fjdhpX jYWTeZ"
  >
    <div />
    <div>
      <input
        name="searchText"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        className="sc-bdVaJa sc-jzJRlG bkSOxo StyledButton-sc-323bzc-0 aNlbA"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        + Issue Credential
      </button>
    </div>
  </section>
</div>
`;
